src/mono/wasm/build/WasmApp.targets

```diff
@@ -140,7 +140,9 @@
 
       The path might not have been created yet, for example when creating a new project in VS, so don't use an Exists() check
     -->
-    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == ''">$([System.IO.Path]::Combine($(OutputPath), 'browser-wasm', 'AppBundle'))</_AppBundleDirForRunCommand>
+    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == '' and '$(TargetFramework)' != '' and $(OutputPath.Contains('browser-wasm'))">$([System.IO.Path]::Combine($(OutputPath), 'AppBundle'))</_AppBundleDirForRunCommand>
+    <_AppBundleDirForRunCommand Condition="'$(_AppBundleDirForRunCommand)' == '' and '$(TargetFramework)' != '' and !$(OutputPath.Contains('browser-wasm'))">$([System.IO.Path]::Combine($(OutputPath), 'browser-wasm', 'AppBundle'))</_AppBundleDirForRunCommand>
```

Comment:
Maybe we don't need the second one? the third one would also do the same thing ?

Also, is this running on customer machines ? What is the impact there ?

Reply:
I was trying to solve it with one option but it's impossible. 
We need `_AppBundleDirForRunCommand ` to be in the form of:
`whatever\Debug\net7.0\browser-wasm\AppBundle`, otherwise the tests/builds won't be passing.
Simultaneously we are experiencing 3 different forms of `OutputPath`, either
- bin\System.Linq.Tests\Debug\net7.0\browser-wasm\ (one framework, libraries, we need to append only AppBundle)
- bin\System.Linq.Expressions.Tests\Debug\ (libraries, multiple frameworks defined, so we need to append the current framework NetCoreAppCurrent and then the rest, browser-wasm + AppBundle)
- wbt\Debug\net7.0\ (one framework, wbt, we need to append both browser-wasm and AppBundle).

What do you mean by customer machines? Wbt is passing in this form.

Reply:
> What do you mean by customer machines? 

We need it to make our tests work. Probably because we have some custom `OutputPath` path logic in our tests.
I imagine that customer's projects could have also some custom path logic, which could be different from our.

1) do we need to make this change and impact customers  or we could solve it for our tests by unifying `OutputPath` there ?
2) if we want or need to make this change with customer impact, would it break some of their custom `OutputPath` setup ?

Reply:
As far as I test, no customer impact.
It's a comparison how a project is behaving without and with this PR. I am setting custom output path:

- `<OutputPath>$(SolutionDir)bin\Debug\net7.0\browser-wasm</OutputPath>`
Without PR:  AppBundle is placed in `bin\Debug\net7.0\browser-wasm\net7.0\browser-wasm`
After applying the PR the value does not change for TargetFrameworks and TargetFramework.
- `<OutputPath>$(SolutionDir)browser-wasm\bin\Debug\net7.0</OutputPath>`
Without PR:  that results in placing my AppBundle in:
`browser-wasm\bin\Debug\net7.0\net7.0\browser-wasm`
After applying the PR the value does not chang  for TargetFrameworks and TargetFramework.
- `<OutputPath>$(SolutionDir)bin\Debug\net7.0</OutputPath>`
 Without PR:  AppBundle is in `bin\Debug\net7.0\net7.0\browser-wasm`
After applying the PR the value does not change for TargetFrameworks and TargetFramework.
- <OutputPath>$(SolutionDir)something</OutputPath>
Without PR: `something\net7.0\browser-wasm`
After applying the PR the value does not change for TargetFrameworks and TargetFramework.